src/main/java/MorseCodePresenter.java

```diff
@@ -0,0 +1,48 @@
+public class MorseCodePresenter
+{
+    private MorseCodeFrame view;
+    private MorseCodeConverter model;
+
+    public MorseCodePresenter(MorseCodeFrame view, MorseCodeConverter model)
+    {
+        this.view = view;
+        this.model = model;
+    }
+
+    public void translateToEnglish(String expression)
+    {
+        if (view.getMorseExpressionText() == null || view.getMorseExpressionText().strip().equals(""))
+        {
+            view.setMorseExpressionText("Make sure to enter some Morse Code before clicking on Translate To English");
+        }
+        else
+        {
+            view.setEnglishExpressionText("");
+            String english = model.toEnglish(expression);
+            view.setEnglishExpressionText(english);
+            if (view.getEnglishExpressionText() == null || view.getEnglishExpressionText().equals(""))
```

Comment:
Instead of getting the text from the view, use the text that you just translated which should be in `english`